### PR TITLE
LPS-145305 Add a sample for FDS `classic-display` tag

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-145305 Create a sample for FDS classic-display tag](https://issues.liferay.com/browse/LPS-145305)

### What is the goal of this PR?

We are adding a showcase of `classic-display` tag, in which FDS can be used without headless API. The key part is implementation of `FDSDataProvider`, where we fetch DXP users through standard `LocalService`.

Notes:
- We cannot set `List<User>` directly, because `User` cannot be serialized. That is why we are using `UserEntry` POJO.
- We are using `_usersAdmin.getUsers` because of the weird signature of `_userLocalService.search`. In its overloads, some return `List<User>` and some `Hits`. The one that is interesting to FDS is [the one including `Sort` argument](https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/service/UserLocalServiceUtil.java#L2092-L2099), and that one returns `Hits`.
- We can extend this tab with other FDS features in future PRs, for the features that have different implementations in two tags. An example for this is sorting, that we are adding is this PR. 
- We can now add CI tests for this tag, starting with simple 'is it rendering' sanity check @Esther-OC 

cc @dsanz 

### What does it look like?

screen

### CODE REVIEW

**Comment by markocikos** [comment]
>1: Do not import internal class from another module: ./modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/provider/ClassicFDSDataProvider.java 21

I am not importing internal class from another module! :thinking:  On L21 is import of an internal class in the same module:
```
import com.liferay.frontend.data.set.sample.web.internal.model.UserEntry;
```

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/view/ClassicTableFDSView.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	@Override
+	public FDSTableSchema getFDSTableSchema(Locale locale) {
+		FDSTableSchemaBuilder fdsTableSchemaBuilder =
+			_fdsTableSchemaBuilderFactory.create();
+
+		return fdsTableSchemaBuilder.add(
+			"firstName", "first-name"
+		).add(
+			"lastName", "last-name",
+			fdsTableSchemaField -> fdsTableSchemaField.setSortable(true)
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/view/ClassicTableFDSView.java:48]: Interesting, lambda expressions. I never played with this before on Java 😂

**Comment by diegonvs** [comment]
Hey @markocikos, what means Hits? Is it like a result from a Search API?

**Review by diegonvs — APPROVED** [review]
NLHTM

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/view/ClassicTableFDSView.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/view/ClassicTableFDSView.java:48]: Yep, it's almost the same. The only difference is that the argument (`fdsTableSchemaField`) cannot be anything, but needs to [match](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-impl/src/main/java/com/liferay/frontend/data/set/internal/view/table/FDSTableSchemaBuilderImpl.java#L45-L61) some interface (`FDSTableSchemaField`).

Java folks are shamelessly copying JS. I recently found out they even [added](https://www.geeksforgeeks.org/var-keyword-in-java/)  `var` for variable type. Next step, I guess, is adding `const` and `let` :joy:

**Comment by markocikos** [comment]
> Hey markocikos, what means Hits? Is it like a result from a Search API?

Yeah, it's a Liferay-specific interface. I am not sure why we are using this, and not just return `List<Users>` always.

But, something interesting I just noticed, `extends Serializable` :thinking: I'll try to simplify solution using this.

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/view/ClassicTableFDSView.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (beltranrengifo) [reply modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/view/ClassicTableFDSView.java:48]: 🤣

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/partials/classic.jsp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<frontend-data-set:classic-display
+	dataProviderKey="<%= FDSSampleFDSNames.CLASSIC %>"
```
Comment (beltranrengifo) [inline modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/partials/classic.jsp:20]: This attribute makes wires the FDS to lookup for a `getItems` method in the provider file? I wonder 🤔

**Review by beltranrengifo — APPROVED** [review]
So nice to see this @markocikos !

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/partials/classic.jsp (hunk 2 above)
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/partials/classic.jsp:20]: Yep. It's is [the same registry mechanism](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-impl/src/main/java/com/liferay/frontend/data/set/internal/provider/FDSDataProviderRegistryImpl.java) that we also have for views and filters.

**Comment by markocikos** [comment]
> But, something interesting I just noticed, extends Serializable thinking I'll try to simplify solution using this.

Reached a dead end with this idea. I tried to set a data provider with `Hits`, but we have `List` in the [generic method](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/provider/FDSDataProvider.java). `Hits` is already a collection, there is no `List<Hits>`.

**Comment by markocikos** [comment]
ci:test:sf

**Comment by markocikos** [comment]
I figured out why CI is failing, I didn't append the `UserEntry` POJO. I think it didn't show up in smartgit at the time, but might have been a brainfart :sweat_smile:  Since there is nothing meaningful to review in that file, and existing reviews are green, I'll forward away.

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**